File: mobile/flutter-pos/lib/data/models/closing.dart
import 'num_util.dart';
class Closing {
  final int idClosing;
  final int idShift;
  final double totalPenjualan;
  final double totalCash;
  final double totalQris;
  final double totalPengeluaran;
  final double saldoAkhir;
  final String? waktuClosing;
  final String status;

  const Closing({
    required this.idClosing,
    required this.idShift,
    required this.totalPenjualan,
    required this.totalCash,
    required this.totalQris,
    required this.totalPengeluaran,
    required this.saldoAkhir,
    this.waktuClosing,
    required this.status,
  });

  factory Closing.fromJson(Map<String, dynamic> json) => Closing(
        idClosing: json['id_closing'] as int,
        idShift: json['id_shift'] as int,
        totalPenjualan: parseDouble(json['total_penjualan']),
        totalCash: parseDouble(json['total_cash']),
        totalQris: parseDouble(json['total_qris']),
        totalPengeluaran: parseDouble(json['total_pengeluaran']),
        saldoAkhir: parseDouble(json['saldo_akhir']),
        waktuClosing: json['waktu_closing'] as String?,
        status: json['status'] as String,
      );
}

File: backend/laravel-api/app/Models/Closing.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Closing extends Model
{
    protected $table = 'closing_kasir';
    protected $primaryKey = 'id_closing';
    public $timestamps = false;

    protected $fillable = [
        'id_shift',
        'total_penjualan',
        'total_cash',
        'total_qris',
        'total_pengeluaran',
        'saldo_akhir',
        'waktu_closing',
        'status',
    ];

    protected $casts = [
        'total_penjualan' => 'decimal:2',
        'total_cash' => 'decimal:2',
        'total_qris' => 'decimal:2',
        'total_pengeluaran' => 'decimal:2',
        'saldo_akhir' => 'decimal:2',
        'waktu_closing' => 'datetime',
    ];

    public function shift(): BelongsTo
    {
        return $this->belongsTo(Shift::class, 'id_shift', 'id_shift');
    }
}
